### PR TITLE
#6775: Integrated widget type around feature with read–only, restricted editing and track changes

### LINKED ISSUE #6775: Read-only support for magic blocks
Pretty self-explanatory 😛

### LINKED ISSUE #6823: Implement the insert paragraph command for the type around feature
## Provide a description of the task

It should be based on `WidgetTypeAround#_insertParagraph` but with additional checks like described in https://github.com/ckeditor/ckeditor5/pull/6794#discussion_r424966959.

### LINKED ISSUE #7229: WidgetTypeAround should use insertContent in order to integrate with track changes
## 📝 Provide a description of the improvement

It uses `writer.insert()` while it should use `model.insertContent()`.

In addition, it should probably copy the implementation of the current \<kbd>Enter\</kbd> handler that we have for widgets. AFAICS it does some more checks.

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (widget): Integrated the `WidgetTypeAround` plugin with the read–only mode. Closes #6775.

Internal (widget): Integrated the `WidgetTypeAround` plugin with the `RestrictedEditingMode` plugin (see #6775).

Feature (paragraph): Implemented the `InsertParagraphCommand` registered as `'insertParagraph'` in the editor. Closes #6823. Closes #7229.

Docs (engine): Improved `Model#insertContent` API documentation (see #6775).

### CODE REVIEW

File: packages/ckeditor5-paragraph/src/insertparagraphcommand.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * **Note**: This command moves the selection to the inserted paragraph.
+ *
+ * @extends module:core/command~Command
+ */
+export default class InsertParagraphCommand extends Command {
+	/**
+	 * Executes the command.
+	 *
+	 * @protected
```
Comment (niegowski) [inline packages/ckeditor5-paragraph/src/insertparagraphcommand.js:29]: Why is this protected, it's public in base class.

Comment (oleq) [reply packages/ckeditor5-paragraph/src/insertparagraphcommand.js:29]: Copy&paste from https://github.com/ckeditor/ckeditor5/blob/i/6775/packages/ckeditor5-highlight/src/highlightcommand.js#L44-L47. Then it's wrong here too :)

File: packages/ckeditor5-paragraph/src/insertparagraphcommand.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			return;
+		}
+
+		editor.model.change( writer => {
+			const paragraph = writer.createElement( 'paragraph' );
+			const selection = writer.createSelection( paragraph, 'in' );
+
+			editor.model.insertContent( paragraph, modelPosition );
+
+			writer.setSelection( selection );
```
Comment (Reinmar) [inline packages/ckeditor5-paragraph/src/insertparagraphcommand.js:57]:
```suggestion
			const paragraph = writer.createElement( 'paragraph' );

			editor.model.insertContent( paragraph, modelPosition );

			writer.setSelection( paragraph, 'in' );
```

Comment (Reinmar) [reply packages/ckeditor5-paragraph/src/insertparagraphcommand.js:57]: And the same needs to be fixed in insertContent docs.

Comment (Reinmar) [reply packages/ckeditor5-paragraph/src/insertparagraphcommand.js:57]: And the same needs to be fixed in insertContent docs.

File: packages/ckeditor5-paragraph/src/insertparagraphcommand.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import Command from '@ckeditor/ckeditor5-core/src/command';
+
+/**
+ * The insert paragraph command. It insert a new paragraph next to the provided model element.
+ *
+ *		// Insert a new paragraph before the first element in the document root.
+ *		editor.execute( 'insertParagraph', {
+ *			element: editor.model.document.getRoot().getChild( 0 ),
+ *			position: 'before'
+ *		} );
```
Comment (Reinmar) [inline packages/ckeditor5-paragraph/src/insertparagraphcommand.js:19]:
```suggestion
 *		editor.execute( 'insertParagraph', {
 *			position: editor.model.createPositionBefore( element )
 *		} );
```

File: packages/ckeditor5-paragraph/src/insertparagraphcommand.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * @module paragraph/insertparagraphcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+
+/**
+ * The insert paragraph command. It insert a new paragraph next to the provided model element.
+ *
+ *		// Insert a new paragraph before the first element in the document root.
```
Comment (Reinmar) [inline packages/ckeditor5-paragraph/src/insertparagraphcommand.js:15]:
```suggestion
 *		// Insert a new paragraph before the given `element`.
```

**Comment by oleq** [comment]
Ready for another round of review, @niegowski.

**Pull request merged**